File: src/main/java/javabot/web/auth/RestrictedProvider.java
package javabot.web.auth;

import com.sun.jersey.api.model.Parameter;
import com.sun.jersey.core.spi.component.ComponentContext;
import com.sun.jersey.core.spi.component.ComponentScope;
import com.sun.jersey.spi.inject.Injectable;
import com.sun.jersey.spi.inject.InjectableProvider;
import javabot.web.auth.Restricted;

public class RestrictedProvider implements InjectableProvider<Restricted, Parameter> {

    @Override
    public ComponentScope getScope() {
        return ComponentScope.PerRequest;
    }

    @Override
    public Injectable<?> getInjectable(ComponentContext ic, Restricted a, Parameter c) {
        return new OpenIDRestrictedToInjectable(a.value());
    }
}

File: src/main/kotlin/javabot/web/auth/RestrictedProvider.kt
package javabot.web.auth

import com.sun.jersey.api.model.Parameter
import com.sun.jersey.core.spi.component.ComponentContext
import com.sun.jersey.core.spi.component.ComponentScope
import com.sun.jersey.core.spi.component.ComponentScope.PerRequest
import com.sun.jersey.spi.inject.Injectable
import com.sun.jersey.spi.inject.InjectableProvider

class RestrictedProvider : InjectableProvider<Restricted, Parameter> {

    override fun getScope(): ComponentScope {
        return PerRequest
    }

    override fun getInjectable(ic: ComponentContext, a: Restricted, c: Parameter): Injectable<*> {
        return OpenIDRestrictedToInjectable(a.value)
    }
}
